account_invoice_negative_total/models/account_invoice.py

```diff
@@ -0,0 +1,50 @@
+from odoo import _, api, models
+from odoo.exceptions import UserError
+from odoo.tools import float_compare
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    @api.multi
+    def action_invoice_open(self):
+        to_open_invoices = self.filtered(lambda inv: inv.state != "open")
+        if self.user_has_groups(
+            "account_invoice_negative_total."
+            "group_validate_invoice_negative_total_amount"
+        ) and to_open_invoices.filtered(
+            lambda inv: float_compare(
+                inv.amount_total,
+                0.0,
+                precision_rounding=inv.currency_id.rounding,
+            )
+            == -1
```

Comment:
This is rather confusing to read to me. I propose something like this:

```suggestion
            lambda inv: not float_is_zero(
                inv.amount_total,
                precision_rounding=inv.currency_id.rounding,
            )
            and inv.amount_total < 0
```